subgraph_extractor/cli.py

```diff
@@ -154,12 +158,20 @@ def convert_columns(df, database_types, table_config):
         field = schema.field(field_index)
         new_field = field.with_type(types[new_type])
         schema = schema.set(field_index, new_field)
-
-    table = table.cast(schema, safe=False)
+    if len(df) == 0:
+        # It's possible that casting an empty table causes issues
+        # but there is a helper on schemas to create a valid
+        # empty table of the same type
+        table = schema.empty_table()
+    else:
+        table = table.cast(schema, safe=False)
     return table
 
 
-def get_partition_iterator(min_partition, max_partition, partition_sizes):
+def get_partitions(min_partition, max_partition, partition_sizes):
```

Comment:
Just stopping this being an iterator as all of the uses now require pulling the whole list.